### PR TITLE
Dynamically change number of formatter buttons in md toolbar overflow menu

### PR DESCRIPTION
**Pull request by aitchiss**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [X] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

We had a couple of issues relating to the approach we took to make the toolbar responsive on smaller screens - previously it scrolled horizontally, but there were issues of discoverability and also usability when the scrolling behaviour was triggered on non-touch devices (e.g. in the moderator centre where we render inside a smaller iframe, but also if a user zoomed their viewport for a11y).

Instead of scrolling in small viewports, we now dynamically detect how many items we place in the overflow (3 dots) menu. See the [figma mockups](https://www.figma.com/file/PIPeiiVSADmcpCbiZTdJm9/Editor?node-id=2185%3A8563).

**A small implementation detail:** 

In extremely small screens (super-narrow smartphones like galaxy fold, or mobile devices where a user is choosing to zoom the screen), I've kept the scrolling behaviour as before. I've done this purely as a fall-back for edge cases like screen zooming for a11y, or those rarer width smartphones. 

The solution we have at the moment is fairly lightweight/understandable, and I didn't want to introduce a ton of complexity just for these less-likely scenarios (but also didn't want things to look terrible for those users).

## Related Tickets & Documents

- Closes https://github.com/forem/forem/issues/17714
- Closes https://github.com/forem/forem/issues/18131

## QA Instructions, Screenshots, Recordings

This needs tested in 3 locations:
- The editor
- Comments on a post
- In the mod centre, commenting on a post

In each case, experiment with resizing your screen and check that everything looks sensible 😄  Please also check some things that gave us headaches before:

- In larger screen sizes, tooltips appear on hover
- Regardless of the overflow menu being open, or the bar being horizontally scrollable, there is never a vertical scroll

https://user-images.githubusercontent.com/20773163/181026057-58a36a2c-c72b-401a-a965-7acdcf9ab687.mp4

### UI accessibility concerns?

I think this will make things a bit more usable for all.

## Added/updated tests?

- [X] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [X] This change does not need to be communicated, and this is why not: small tweak, no new functionality

### CODE REVIEW

File: app/assets/stylesheets/views/article-form.scss [hunk 1]
```diff
@@ -158,32 +158,26 @@
...
-    overflow-x: auto;
+    padding-right: var(--content-padding-x, 0);
     flex-shrink: 0;
     margin: calc(var(--content-padding-y) * -1)
       calc(var(--content-padding-x) * -1) var(--su-6)
       calc(var(--content-padding-x) * -1);
 
     .editor-toolbar {
-      position: relative;
+      overflow-x: auto;
```
Comment (aitchiss) [inline app/assets/stylesheets/views/article-form.scss:168]: As mentioned in the PR description, this is purely a fallback for edge-cases, and we remove this property at our XS breakpoint below

**Comment by Link2Twenty** [comment]
This might be a totally crazy idea but what do you think of something like this?

https://jsfiddle.net/link2twenty/b0n2pgjz/

**Comment by Link2Twenty** [comment]
> We could even use a ResizeOberserver on the toolbar itself

I didn't even know that was a thing 😅
I've updated my jsfiddle to use that now.

**Review by fdocr — APPROVED** [review]
Ran through all three locations and the tooltip to show the extra buttons works nicely!

The only issue was using the Mod center on a mobile device (small screen) but that's to be expected I believe because of the layout within the iframe isn't wide enough. The fallback scroll appears though so that's good 😄

**Review by joshpuetz — COMMENTED** [review]
Tested locally and works great with one exception (which I'm not sure if it's a bug or not):

In the mod center, below about 645 pixels the three dots in the menu bar for comment scroll out of view:
CleanShot 2022-07-27 at 11 29 31

**Review by joshpuetz — APPROVED** [review]

**Comment by aitchiss** [comment]
> Tested locally and works great with one exception (which I'm not sure if it's a bug or not)

Yeah the mod centre is not optimised for small screens at all 😅 In this small viewport, the horizontal scroll fallback kicks in for the toolbar, but yeah it's not 100% ideal. I think this could be improved in a later iteration with @Link2Twenty's suggestion, but I'll merge this for now since it gets us _most_ of the way there 👍

**Comment by thomasbnt** [comment]
So cool ! Thanks @aitchiss 🎉

**Comment by andypiper** [comment]
Thank you @aitchiss this is super helpful!

**Comment by Link2Twenty** [comment]
The react based POC I said I was going to make is ready.
https://codesandbox.io/s/angry-cloud-k2ucr4?file=/src/components/DynamicToolbar/DynamicToolbar.jsx

**Pull request merged**